src/sentry/preprod/api/endpoints/project_preprod_check_for_updates.py

```diff
@@ -46,22 +55,125 @@ def get(self, request: Request, project) -> Response:
         except PreprodArtifact.DoesNotExist:
-            return Response({"error": "Not found"}, status=404)
+            logger.warning(
+                "No artifact found for binary identifier with version %s", provided_version
+            )
 
-        if preprod_artifact.build_version and preprod_artifact.build_number:
+        if preprod_artifact and preprod_artifact.build_version and preprod_artifact.build_number:
             current = InstallableBuildDetails(
+                id=str(preprod_artifact.id),
                 build_version=preprod_artifact.build_version,
                 build_number=preprod_artifact.build_number,
+                app_name=preprod_artifact.app_name,
+                download_url=get_download_url_for_artifact(preprod_artifact, request),
+                created_date=preprod_artifact.date_added.isoformat(),
             )
-        else:
-            current = None
-        return Response(CheckForUpdatesApiResponse(current=current).dict())
+
+        # Get the update object - find the highest version available
+        # Get all build versions for this app and platform
+        filter_kwargs = get_base_filters()
+        if preprod_artifact:
+            filter_kwargs["build_configuration"] = preprod_artifact.build_configuration
+        all_versions = (
+            PreprodArtifact.objects.filter(**filter_kwargs)
+            .values_list("build_version", flat=True)
+            .distinct()
+        )
+
+        # Find the highest semver version
+        highest_version = None
+        for version in all_versions:
+            if version:
+                try:
+                    parsed_version = parse_version(version)
+                    if highest_version is None or parsed_version > parse_version(highest_version):
+                        highest_version = version
+                except Exception:
+                    # Skip invalid version strings
+                    continue
+
+        # Get all artifacts for the highest version
+        if highest_version:
+            filter_kwargs = get_base_filters()
+            filter_kwargs["build_version"] = highest_version
+            potential_artifacts = PreprodArtifact.objects.filter(**filter_kwargs)
+
+            # Filter for installable artifacts and get the one with highest build_number
+            installable_artifacts = [
+                artifact for artifact in potential_artifacts if is_installable_artifact(artifact)
+            ]
+            if len(installable_artifacts) > 0:
+                build_numbers = [artifact.build_number for artifact in installable_artifacts]
+                installable_artifacts = [
+                    artifact
+                    for artifact in installable_artifacts
+                    if artifact.build_number == max(build_numbers)
+                ]
+                # Get the artifact with the highest creation date
+                best_artifact = max(installable_artifacts, key=lambda x: x.date_added)
```

Comment:
I think this can be simplified

```suggestion
                best_artifact = max(installable_artifacts, key=lambda a: (a.build_number, a.date_added))
```